src/orange/orangeinp/detail/CsgUnitBuilder.cc

```diff
@@ -193,6 +194,47 @@ void CsgUnitBuilder::fill_volume(LocalVolumeId v,
     CELER_ENSURE(is_filled(unit_->fills[v.unchecked_get()]));
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Simplify negated joins for Infix evaluation.
+ *
+ * Apply DeMorgan simplification to use the \c CsgUnit in infix evaluation.
+ * \c NodeId indexing in the \c CsgTree are invalidated after calling this,
+ * \c CsgUnit data is updated to point to the simplified tree \c NodeId but any
+ * previously cached \c NodeId is invalid.
+ */
+void CsgUnitBuilder::simplifiy_joins()
+{
+    auto& tree = unit_->tree;
+    auto simplification = transform_negated_joins(tree);
+    CELER_EXPECT(tree.size() == simplification.new_nodes.size());
+    std::vector<std::set<CsgUnit::Metadata>> md;
+    md.resize(simplification.tree.size());
+
+    std::map<NodeId, CsgUnit::Region> regions;
+
+    for (auto node_id : range(tree.size()))
+    {
+        if (auto equivalent_node = simplification.new_nodes[node_id];
+            equivalent_node)
+        {
+            CELER_EXPECT(equivalent_node < md.size());
+            md[equivalent_node.unchecked_get()] = unit_->metadata[node_id];
+            regions[equivalent_node] = unit_->regions[NodeId{node_id}];
```

Comment:
Can we `move` them? (Nodes in the CSG tree shouldn't get duplicated, so this should be one-to-one right?)
```suggestion
            md[equivalent_node.unchecked_get()] = std::move(unit_->metadata[node_id]);
            regions[equivalent_node] = std::move(unit_->regions[NodeId{node_id}]);
```

Reply:
yep should be fine to move since we reassign `unit_->metadata` and `unit_->regions` later.